packages/ng/src/app/select/searcher/select-searcher.component.ts

```diff
@@ -0,0 +1,132 @@
+	providers: [{provide: AbstractSelectOptionFeederComponent, useExisting: forwardRef(() => LuSelectSearcherComponent)}]
+})
+/**
+ * Component that manage the possibility to search in the options of a select.
+ */
+export class LuSelectSearcherComponent<T> extends AbstractSelectOptionFeederComponent<T> implements ISelectSearcher<T>, OnDestroy  {
+
+	_clue = '';
+	private _focus = false;
+	_noResults = false;
+	private _clue$: Subject<string> = new Subject<string>();
+
+	private _intlChanges: Subscription;
+
+	private innerMap = {};
+
+	/**
+	 * The options detected
+	 */
+	@ContentChildren(LuSelectOption, { descendants: true }) luOptions: QueryList<LuSelectOption<T>>;
+	@ViewChild('inputClue') _inputElement : ElementRef;
+	@ViewChild('scrollElement') _scrollElement : ElementRef;
+
+
+	constructor(public _intl: LuSelectSearchIntl,
+		private _changeDetectorRef: ChangeDetectorRef) {
+		super();
+		this._clue$
+		.debounceTime(100) // wait 100ms after the last event before emitting last event
+		.distinctUntilChanged() // only emit if value is different from previous value
+		.subscribe(model => {
+			this._clue = model;
+			// Call the filter function
+			this._noResults = this.filter(this._clue, this.luOptions.toArray()).length === 0;
+		});
+		this._intlChanges = _intl.changes.subscribe(() => this._changeDetectorRef.markForCheck());
+	}
+
+	ngOnDestroy() {
+		this._intlChanges.unsubscribe();
+	}
+
+
+	_onMouseDown($e) {
+		this._focus = true;
+		// We prevent propagation to avoid lost of focus in input field
+		$e.stopPropagation();
+	}
+
+	_onBlur() {
+		this._focus = false;
+		// When we quit the field, we reset the search item
+		this._clue = '';
+		this.luOptions.toArray().forEach(luOption => luOption.displayed = true);
+	}
+
+	_onKeydown($event: KeyboardEvent){
+		this._callbackKeyEvent($event);
+	}
+
+	/**
+	 * See ISelectSearcher
+	 */
+	filter(clue: string, options: LuSelectOption<T>[]): LuSelectOption<T>[] {
+		const normalizeClue = clue.normalize('NFD').replace(/[\u0300-\u036f]/g, "").toLowerCase();
+		return options.filter((option) => {
+			let valueOption = this.innerMap[option.viewValue];
+			if (!valueOption){
+				this.innerMap[option.viewValue] = option.viewValue.normalize('NFD').replace(/[\u0300-\u036f]/g, "").toLowerCase();
+				valueOption = this.innerMap[option.viewValue];
+			}
+
+			const match = valueOption.indexOf(normalizeClue) !== -1;
+			option.displayed = match;
+			return match;
+		});
+	}
+
+	/**
+	 * See ISelectOptionFeeder
+	 */
+	hasFocus(): boolean {
+		return this._focus;
+	}
+
+	_onFieldChange(clue: string) {
+		this._clue$.next(clue);
+	}
+
+	/**
+	 * See ISelectOptionFeeder
+	*/
+	open(): void {
+		this._focus = true;
+		this._inputElement.nativeElement.focus();
+	}
+
+	/**
+	 * See ISelectOptionFeeder
+	 */
+	scrollTo(index: number) {
+		const luOption = this.luOptions.toArray()[index];
+
+		this._scrollElement.nativeElement.scrollTop = luOption.offsetTop();
+
+	}
```

Comment:
does this method scroll to the nth option or the nth displayed option ?

Reply:
Yep :) because, it use the ElementRef and nativeElement so, il reference the real offsetTop of the element

Reply:
you didn't answer me. I was asking if it goes to the nth option, or the nth displayed option (in case some are filtered because of whats inside the search).

in any case you can have a method `scrollTo(option) instead of `scrollTo(index)`. in the first synthax it doesn't matter if option is displayed or not

Reply:
No I need the index number, because, i use the option, i need to search again the option in the list. If I use number instead of Option it's to avoid an other search in the list. 

And if the list is filtered (with the search for example) all the index are up to date so image this use case : 
* I Have 1000 options and the popup is open. 
* The item selected is the 790
* I use the key down to navigate in the list 
* It will scroll directly to the 790 items

Now i will start to filter the items : 
* I write a text that remove 700 items so I still have a list with 300 items and the new index of my selected value is now 50
* When i use the key down, it will be place correctly to index 50 and not 790

I hope it anwser to your questions

Reply:
well you need to be sure the index is right whenever you use the method, not just in the current implementation

in both cases what i heard you say is _i want to scroll to the selected option_. Do you have complete trust in the code making sure the index you use is the right one ? 

finding an option in a list complexity is `O(n)` at worst, and its probable that third party libs optimise better, so it doesn't matter if you search again.

I'd rather have a clean code to an optimised one, especially for the first draft, when we dont even know if optimising this is pertinent